### PR TITLE
Mocha to jest migration for eslint packages

### LINKED ISSUE #10709: ESLint-related packages migration - TODO
- [x] Merge https://github.com/babel/babel/pull/10705
- [x] Remove lock files from new packages (https://github.com/babel/babel/pull/10711)
- [ ] ~~Ignore new packages while linting~~
- [x] Make "Lint" pass on CI (https://github.com/babel/babel/pull/10712)
- [x] Move CHANGELOG files to `.github/` (https://github.com/babel/babel/pull/10714)
- [x] Actually rename the new packages (https://github.com/babel/babel/pull/10715)
- [x] Prevent lerna from publishing those packages until we are ready (@nicolo-ribaudo) (https://github.com/babel/babel/pull/10722)
- [x] Unify build process (https://github.com/babel/babel/pull/10723, https://github.com/babel/babel/pull/10729)
- [x] Migrate from `mocha` to `jest` (which is used by the other `@babel/` packages) (https://github.com/babel/babel/pull/10716)
- [x] Disable tests for ESLint related packages on unsupported node versions (https://github.com/babel/babel/pull/10730)

### PR DESCRIPTION
**Pull request by rajasekarm**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes partially #10709 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

- [x] babel-eslint-parser
- [x] babel-eslint-plugin
- [x] babel-eslint-plugin-development

### CODE REVIEW

File: eslint/babel-eslint-parser/package.json [hunk 1]
```diff
@@ -13,7 +13,6 @@
   },
   "homepage": "https://github.com/babel/babel-eslint",
   "scripts": {
-    "test": "cd test && mocha specs && cd -",
     "preversion": "npm test",
```
Comment (JLHwung) [inline eslint/babel-eslint-parser/package.json]: The `preversion` hook is also unnecessary now.

File: eslint/babel-eslint-parser/test/integration.js [hunk 2]
```diff
@@ -1,16 +1,15 @@
 "use strict";
 
-const assert = require("assert");
 const eslint = require("eslint");
 const fs = require("fs");
 const path = require("path");
 
-const parser = require("../..");
+const parser = require("../lib");
```
Comment (JLHwung) [inline eslint/babel-eslint-parser/test/integration.js]: Nit: it suffices to use `..` and resolve using metadata (e.g. `main`) in `package.json`.

File: eslint/babel-eslint-parser/test/non-regression.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ const eslint = require("eslint");
 const path = require("path");
 const unpad = require("dedent");
 
-const parser = require("../..");
+const parser = require("../lib");
```
Comment (JLHwung) [inline eslint/babel-eslint-parser/test/non-regression.js]: ditto.

File: eslint/babel-eslint-parser/test/non-regression.js [hunk 4]
```diff
@@ -35,6 +41,8 @@ function verifyAndAssertMessagesWithSpecificESLint(
     }
   }
 
+  // console.log(config);
```
Comment (nicolo-ribaudo) [inline eslint/babel-eslint-parser/test/non-regression.js]: Nit: remove

**Comment by kaicataldo** [comment]
PR adding the build step to the `eslint/*` packages here: https://github.com/babel/babel/pull/10723

File: eslint/babel-eslint-parser/test/integration.js [hunk 5]
```diff
@@ -218,8 +217,10 @@ function strictSuite() {
         },
         (err, report) => {
           if (err) return done(err);
-          assert(report[0].ruleId === ruleId);
-          assert(report[0].nodeType.indexOf("Function") === -1);
+          // assert(report[0].ruleId === ruleId);
```
Comment (JLHwung) [inline eslint/babel-eslint-parser/test/integration.js]: nit: good to remove `assert`-leading comments.

**Pull request merged**